extensions/podman/src/extension.spec.ts

```diff
@@ -1088,3 +1088,33 @@ test('checkDisguisedPodmanSocket: runs updateWarnings when called not on Linux',
   await checkDisguisedPodmanSocket(provider);
   expect(updateWarningsMock).toBeCalled();
 });
+
+test('Even with getJSONMachineList erroring, do not show setup notification on Linux', async () => {
+  vi.mocked(isLinux).mockReturnValue(true);
+  vi.spyOn(extensionApi.process, 'exec').mockRejectedValue({
+    name: 'name',
+    message: 'description',
+    stderr: 'error',
+  });
+  await expect(extension.updateMachines(provider)).rejects.toThrow('description');
+  expect(extensionApi.window.showNotification).not.toBeCalled();
+});
+
+test('If machine list is empty, do not show setup notification on Linux', async () => {
+  vi.mocked(isLinux).mockReturnValue(true);
+  const spyExecPromise = vi.spyOn(extensionApi.process, 'exec');
+  spyExecPromise.mockImplementation(() => {
+    return Promise.resolve({ stdout: '[]' } as extensionApi.RunResult);
+  });
+  await extension.updateMachines(provider);
+  expect(extensionApi.window.showNotification).not.toBeCalled();
+});
+
+test('if there are no machines, make sure checkDefaultMachine is not being ran inside updateMachines', async () => {
+  const spyCheckDefaultMachine = vi.spyOn(extension, 'checkDefaultMachine');
+  vi.spyOn(extensionApi.process, 'exec').mockImplementation(() => {
+    return Promise.resolve({ stdout: '[]' } as extensionApi.RunResult);
+  });
```

Comment:
`mockResolvedValue({stdout: []})` instead of a mockImplementation that return a Promise.resolve

Reply:
updated this test and the other test with your suggestion